compiler-plugin/src/main/kotlin/arrow/meta/phases/CompilerContext.kt

```diff
@@ -10,6 +10,10 @@ import org.jetbrains.kotlin.descriptors.ModuleDescriptor
 import org.jetbrains.kotlin.psi.KtFile
 import org.jetbrains.kotlin.psi.KtPsiFactory
 
+/**
+ * The Compiler Context represents the environment received by all plugins.
+ * The Compiler Context will get more services as they become relevant overtime to the development of compiler plugins.
+ */
```

Comment:
I think it might be good to indicate how it's a wrapper for the context (and to explain the `backendContext` is part of the actual Kotlin compiler)